design/proposals/load_policy/load_policy_validation.md

```diff
@@ -0,0 +1,148 @@
+
+### User Interface
+TBD
+
+## Design
+The following diagrams, represent a new policy validation layer which will be triggered as part of the load policy request,
+As the final state after the current logic.
+
+
+### Flow Diagrams
+In our use case, Host with JWT authenticator annotations validation:
+1. Conjur admin define JWT authenticator 
+2. Conjur admin load a policy of Host to be authenticated with JWT authenticator
+3. New policy validation layer is triggered and validate if the host is configured correctly 
+4. Host authenticate using JWT authenticator, due to above validation there are no configuration errors at the authenticate request stage
+
+### Sequence Diagrams
+![Load policy validation sequence diagram](load-policy-validation-sequence-diagram.png)
+
+#### Sequence Diagrams - additional info
+1. Policy validation entry point:
+As described in the diagram the validation will be triggered as the final new stage of the current logic (read below),
+   which will call to a new command class `ValidatePolicyConstraints` with tne input of the policy entities 
+   records (see example in the end the of the diagram) from the user body request.
+   
+Load policy current logic - 
+   for the load policy rest requests put/post/patch the current logic in high level is:
+    The current logic: 
+       
+- policy syntax check
+- load records from policy 
+- create temp schema
+- merge new temp schema to main schema 
+- the validation logic will start after the schema merged to main
+
+2. new ValidatePolicyConstraints command class - is responsible to validate all the policy entities 
+   (host,user,webservice) which were sent by the user in load policy request.
+   
+    Note: in this design we will validate only hosts 
+   
+3. new CreatePolicyEntityValidatorsFactory command class - is responsible to create a list of validators as the amount 
+   of the different policy entities types in the records input.
+   
+    Note: in this design we will create only `ValidateHostConstraints` with all the hosts ids in his belly 
+   
+4. new ValidateHostConstraints command class - is responsible to validate a list of host ids which are initialized in his creation.
+   the flow is to fetch the host annotations from the DB (from the main schema after the merge) and to 
+   use the `CreateHostValidatorsFactory` to create hosts authenticators validators.
+   
+5. new CreateHostValidatorsFactory command class - is responsible to create a list of validators as the amount
+   of the different hosts authenticators types
+   
+    Note: 
+    - in this design we will create only `ValidateJwtHostConstraints` which is validator for hosts with JWT annotations
+    - this class can be refactored to use reflection if in the future we will want the separate the authenticator logic 
+    to a different gems 
+      
+6. new GetJwtHosts command class - is responsible to get a lists of host and annotations and to return only the hosts wjich are 
```

Comment:
a list of hosts and their annotations
wjich -> which
Can you explaining what is the module that runs the GetJwtHosts or all the functions you describe here please?

Reply:
In high level: in the flow i can get 10+ hosts in 1 policy fro the users as input.
i need to be able to create the Validator which will be know hot to handle each hosts based on his annotations keys (currently we are supporting only JWT) .

So GetJwtHosts helps me to filter the Hosts which have JWT annotations from the list